core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -195,7 +195,7 @@ protected boolean doEquals(QB other) {
 
     @Override
     public final int hashCode() {
-        return 31 * Objects.hash(getClass(), queryName, boost) + doHashCode();
+        return Objects.hash(getClass(), queryName, boost, doHashCode());
```

Comment:
yea I think it makes no difference, but this way we should be more compatible with Objects.hash as we use it.